# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../djv/djv-0.8.2-gcc44.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix compiling with gcc >= 4.4

diff -ruN djv-0.8.2-orig/bin/softimage-flip/flip.cxx djv-0.8.2/bin/softimage-flip/flip.cxx
--- djv-0.8.2-orig/bin/softimage-flip/flip.cxx	2009-05-25 22:46:34.000000000 +0200
+++ djv-0.8.2/bin/softimage-flip/flip.cxx	2011-04-15 14:34:14.008078153 +0200
@@ -10,6 +10,7 @@
 #include <sstream>
 #include <iomanip>
 #include <fstream>
+#include <cstdlib>
 
 using namespace std;
 
diff -ruN djv-0.8.2-orig/lib/djv_base/error.cxx djv-0.8.2/lib/djv_base/error.cxx
--- djv-0.8.2-orig/lib/djv_base/error.cxx	2009-07-05 20:51:48.000000000 +0200
+++ djv-0.8.2/lib/djv_base/error.cxx	2011-04-15 14:11:38.280077537 +0200
@@ -6,6 +6,8 @@
  Use is subject to license terms.
 ------------------------------------------------------------------------------*/
 
+#include <cstdio>
+
 #include "error.h"
 
 #include <djv_base/string.h>
diff -ruN djv-0.8.2-orig/lib/djv_base/file_seq.cxx djv-0.8.2/lib/djv_base/file_seq.cxx
--- djv-0.8.2-orig/lib/djv_base/file_seq.cxx	2009-06-14 22:32:04.000000000 +0200
+++ djv-0.8.2/lib/djv_base/file_seq.cxx	2011-04-15 14:12:43.772077819 +0200
@@ -9,6 +9,8 @@
 #undef DJV_DEBUG
 //#define DJV_DEBUG
 
+#include <cstdio>
+
 #include "file.h"
 
 #include <algorithm>
diff -ruN djv-0.8.2-orig/lib/djv_base/string_fnc.cxx djv-0.8.2/lib/djv_base/string_fnc.cxx
--- djv-0.8.2-orig/lib/djv_base/string_fnc.cxx	2009-06-13 04:53:38.000000000 +0200
+++ djv-0.8.2/lib/djv_base/string_fnc.cxx	2011-04-15 14:13:53.452078287 +0200
@@ -9,6 +9,8 @@
 #undef DJV_DEBUG
 //#define DJV_DEBUG
 
+#include <cstdio>
+
 #include "string.h"
 
 #include <djv_base/math.h>
diff -ruN djv-0.8.2-orig/lib/djv_image/glx_offscreen_pbuffer.cxx djv-0.8.2/lib/djv_image/glx_offscreen_pbuffer.cxx
--- djv-0.8.2-orig/lib/djv_image/glx_offscreen_pbuffer.cxx	2009-05-25 22:46:34.000000000 +0200
+++ djv-0.8.2/lib/djv_image/glx_offscreen_pbuffer.cxx	2011-04-15 14:17:08.460077819 +0200
@@ -8,6 +8,8 @@
 
 //#define DJV_DEBUG
 
+#include <cstdio>
+
 #include "glx_offscreen_private.h"
 
 #include "glx_private.h"
diff -ruN djv-0.8.2-orig/lib/djv_image/io_lut_etc.cxx djv-0.8.2/lib/djv_image/io_lut_etc.cxx
--- djv-0.8.2-orig/lib/djv_image/io_lut_etc.cxx	2009-06-13 04:35:06.000000000 +0200
+++ djv-0.8.2/lib/djv_image/io_lut_etc.cxx	2011-04-15 14:16:13.556077838 +0200
@@ -6,6 +6,8 @@
  Use is subject to license terms.
 ------------------------------------------------------------------------------*/
 
+#include <cstdio>
+
 #include "io_lut.h"
 
 #include <djv_image/color.h>
diff -ruN djv-0.8.2-orig/lib/djv_image/io_ppm_save.cxx djv-0.8.2/lib/djv_image/io_ppm_save.cxx
--- djv-0.8.2-orig/lib/djv_image/io_ppm_save.cxx	2009-06-13 03:06:30.000000000 +0200
+++ djv-0.8.2/lib/djv_image/io_ppm_save.cxx	2011-04-15 14:15:02.692075750 +0200
@@ -8,6 +8,8 @@
 
 //#define DJV_DEBUG
 
+#include <cstdio>
+
 #include "io_ppm.h"
 
 #include <djv_image/gl_data.h>
